### PR TITLE
Refs Task#384078  remove chatbot Unit test

### PR DESCRIPTION
**Pull request by quanghung97**
## Related Tickets

- [#384078](https://dev.sun-asterisk.com/issues/384078)

## What's this PR do ?

- [x] Unit test remove chatbot

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]
ok

File: tests/Feature/ChatbotControllerTest.php [hunk 1]
```diff
@@ -27,4 +27,38 @@ public function testShowListChatbotFeature()
...
     }
+
+    /**
+     * test Feature remove bot successfully.
+     *
+     * @return void
+     */
+    public function testRemoveChatbotFeature()
+    {
+        $chatbotList = factory(Bot::class)->create(['name' => 'test remove bot']);
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: `$chatbotList` viết thành $chatbot or $bot là được rồi chú

File: tests/Feature/ChatbotControllerTest.php [hunk 2]
```diff
@@ -27,4 +27,38 @@ public function testShowListChatbotFeature()
...
+    {
+        $chatbotList = factory(Bot::class)->create(['name' => 'test remove bot fail']);
+        $user = $chatbotList->user;
+
+        $this->actingAs($user);
+        $response = $this->delete(route('bots.destroy', ($chatbotList->id + 99)));
+        $this->assertDatabaseHas('bots', ['name' => 'test remove bot fail']);
+        $response->assertRedirect();
+        $response->assertStatus(302);
+    }
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: viết thêm case user unauthorized nữa chú

File: tests/Feature/ChatbotControllerTest.php [hunk 3]
```diff
@@ -27,4 +27,38 @@ public function testShowListChatbotFeature()
...
+
+    /**
+     * test Feature remove bot fail.
+     *
+     * @return void
+     */
+    public function testRemoveChatbotFailFeature()
+    {
+        $chatbotList = factory(Bot::class)->create(['name' => 'test remove bot fail']);
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: tương tự ở đây nha

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr></table>

**Review by tamtt-1031 — APPROVED** [review]

File: tests/Feature/ChatbotControllerTest.php [hunk 4]
```diff
@@ -27,4 +25,52 @@ public function testShowListChatbotFeature()
...
+     */
+    public function testRemoveChatbotFailFeature()
+    {
+        $bot = factory(Bot::class)->create(['name' => 'test remove bot fail']);
+        $user = $bot->user;
+
+        $this->actingAs($user);
+        $response = $this->delete(route('bots.destroy', ($bot->id + 99)));
+        $this->assertDatabaseHas('bots', ['name' => 'test remove bot fail']);
+        $response->assertRedirect();
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: assertRedirect ko có gì hả chú.
nếu nó vẫn giữ nguyên url thì chú có thể dùng assertLocation

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: tests/Feature/ChatbotControllerTest.php [hunk 5]
```diff
@@ -27,4 +25,52 @@ public function testShowListChatbotFeature()
...
+        $response->assertRedirect('/bots');
+        $response->assertStatus(302);
+    }
+
+    /**
+     * test Feature remove bot fail without middleware
+     *
+     * @return void
+     */
+    public function testStoreWebhookFailFeature()
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: `testStoreWebhookFailFeature` -> remove thì đúng hơn chú nhỉ?

File: tests/Feature/ChatbotControllerTest.php [hunk 6]
```diff
@@ -27,4 +25,52 @@ public function testShowListChatbotFeature()
...
+    {
+        $this->withoutMiddleware();
+        $response = $this->delete(route('bots.destroy', 1));
+
+        $response->assertRedirect('/bots');
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: unauthorized thì ở đây /login mới đúng chứ nhỉ

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**